### PR TITLE
Collapse repeat frames

### LINKED ISSUE #4078: Please truncate/trim stack traces with a large number of repeating calls
E.g.:

image

I have literally pages and pages full of this because there was a RangeError with the max call stack exceeding. This large number of lines in the stack trace slows down the app a lot.

Is it possible to have Sentry detect these kind of repeating stack traces and truncate them with a message saying something like "Truncated (2300 more)"?

### PR DESCRIPTION
**Pull request by MaxBittker**
filter repeated stack frames in UI
this works by comparing if two stack frames share a lineNo and function name. displays a small card eg: (↻ 20248)
fixes #4078
/cc @getsentry/product

screenshot 2016-09-27 15 00 20

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4224)

### CODE REVIEW

**Comment by MaxBittker** [comment]
screenshot 2016-09-27 15 37 28

**Comment by ckj** [comment]
@MaxBittker 😍 . Colors and position both feel pretty good. 👍  from me.

**Comment by benvinegar** [comment]
Pretty cool.

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceContent.jsx [hunk 1]
```diff
@@ -77,9 +86,15 @@ const StacktraceContent = React.createClass({
             emptySourceNotation={lastFrameIdx === frameIdx && frameIdx === 0}
             isOnlyFrame={this.props.data.frames.length === 1}
             nextFrameInApp={nextFrame && nextFrame.inApp}
-            platform={this.props.platform} />
+            platform={this.props.platform}
+            timesRepeated={nRepeats}/>
         );
       }
+
+      if(!repeatedFrame){
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceContent.jsx]: One space after `if` and one space before the brace:

```javascript
if (!repeatedFrame) {
```

**Review by benvinegar — CHANGES_REQUESTED** [review]
This looks great – my only continued feedback is to change the style so that it matches most of the other code.

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceContent.jsx (hunk 1 above)
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceContent.jsx]: i got this

**Review by benvinegar — APPROVED** [review]

**Pull request merged**